packages/@sanity/types/src/schema/types.ts

```diff
@@ -450,7 +451,18 @@ export interface ConditionalPropertyCallbackContext {
 export type ConditionalPropertyCallback = (context: ConditionalPropertyCallbackContext) => boolean
 export type ConditionalProperty = boolean | ConditionalPropertyCallback | undefined
 
-export type InitialValueResolver<Params, Value> = (params?: Params) => Promise<Value> | Value
+export interface SourceContext {
```

Comment:
Hm, just wondering if `SourceContext` can be confused with `Source` (as returned by `useSource()`).

Could this be named `InitialValueResolverContext` for more clarity?

```suggestion
export interface InitialValueResolverContext {
```

Reply:
Sure. It is the same fields as ConfigContext but that type is in sanity package only afaict.